### PR TITLE
Add capability to only test futures that have skipifs

### PR DESCRIPTION
**Pull request by ronawho**
For a while we've wanted a testing mode where futures are run if and only if
there's a skipif for the test. Historically there was no middle ground and we
either ran 0 or all ~800 futures. We turned futures off for most configurations
since we didn't want to waste time running them everywhere with little reward.
However, some futures are platform specific and we would never have noticed if
they started passing.

This allows us to test configuration specific futures, without have to run all
the futures. There are only ~60 future tests that also have .skipifs. I would
guess that maybe 5-30 futures will run for any given configuration so it
shouldn't add much overhead to testing.

The current default for nightly is to only test futures with skipifs. You can
toss -futures to test all futures, or -no-futures to skip all futures.

Only the "standard" linux64 configuration is tossing -futures for now. I'll
ping the group to see what other configurations we might want to add. I thought
about gasnet-everything and darwin, but wasn't sure.

I was originally thinking that -no-futures wouldn't be used much if at all now.
I removed all uses of it, but now I'm wondering if it makes sense to use it for
verify and memleaks still.

### CODE REVIEW

**Comment by ronawho** [comment]
@thomasvandoren mind reviewing this one?

@bradcray what configurations do we want to test all futures with? Right now I just have them enabled for the "base" linux64 (everything else defaults to futures-skipif.) I thought about gasnet-everthing and darwin but wasn't sure. Are there any we still want to disable all futures for? I think we might want to for memleaks?

File: util/test/sub_test [hunk 1]
```diff
@@ -1023,10 +1021,23 @@ for testname in testsrc:
...
-    # Skip non-future tests if specified
-    if (testfuturesfile==False and testfutures==2):
+
+    # 0: test no futures
+    if testfutures == 0 and testfuturesfile == True:
+        sys.stdout.write('[Skipping future test: %s/%s]\n'%(localdir,test_filename))
+        continue # on to next test
+    # 1: test all futures
+    elif testfutures == 1:
+      pass
```
Comment (thomasvandoren) [inline util/test/sub_test]: I think this line wants to be indented two more spaces.

**Comment by thomasvandoren** [comment]
The changes look good :+1: 

My inclination is to only run all the futures on linux64, at least to start with. But I could be easily talked into running them for other configurations...

**Comment by bradcray** [comment]
I'd be tempted to keep at least one gasnet run doing futures just because I think it's sufficiently different from linux64 that we'll catch some things.  Could go either way on darwin.  No idea about what else we should skip testing all futures on.  I think "none" would be a nice place to be if possible.

**Comment by ronawho** [comment]
I think doing futures for gasnet makes sense. I was thinking gasnet-everything, which has the downside of only running every other day.

I wasn't sure if futures would screw up any leak counts or something for memleaks, otherwise I agree starting with no configs turning off futures would be nice.

**Comment by ronawho** [comment]
@hildeth any thoughts on if memleaks should run with --no-futures: 5d86befc7a38771ee910548c885e6e896acaa93e

If it won't screw up leak counts or anything (I don't actually know how mem leaks testing works) then there's no need to throw it. I just wasn't sure if that commit above was to save time or because futures mess with the results.

**Comment by hildeth** [comment]
Hi Elliot:

I think I stumbled across this when I had futures set in my environment, or just tossed both flags by mistake.

Since the memleaks tests normally do not include futures, it would make the results incomparable if one did so inadvertently (like me).  So I put the --no-futures in there as
 1) documentation, and
 2) a nice safety to keep others from falling in the same trap.

THH
________________________________

**Comment by ronawho** [comment]
So does that mean throw --no-futures so it's easier to do a memleaks run manually?

**Comment by bradcray** [comment]
It sounds to me like it means throw --no-futures so that it doesn't introduce a discontinuity into the results.  Having heard the thought process about that, I'm good with it.

**Comment by ronawho** [comment]
Ah ok, got it. And gasnet-everything (every other day) is ok? Or does gasnet.fifo [also gn-everything] (runs daily) make more sense?

**Comment by bradcray** [comment]
Since we care more about gasnet-everything, I think I'd go with that.  I don't see the every other day as a huge deal.  We could add it to gasnet-fast in addition if it ended up being one.

**Pull request merged**